ClassesObjects/Lab/CatalogVehicleTests/Classes/CatalogVehicle_Test.cs

```diff
@@ -0,0 +1,158 @@
+
+            try
+            {
+                catalog.PrintVehicles();
+                Assert.IsTrue(true);
+            }
+            catch
+            {
+                Assert.IsTrue(false);
+            }
+        }
+
+        [Test]
+        public void Test_PrintVehiclesWithListParameter()
+        {
+            Car newCar1 = new Car("Car", "Mercedes", "CLK500", "250");
+            Car newCar2 = new Car("Car", "Audi", "A3", "90");
+            Car newCar3 = new Car("Car", "Opel", "Astra", "150");
+            Car newCar4 = new Car("Car", "BMW", "M5", "350");
+
+            CatalogueVehicle catalog = new CatalogueVehicle();
+            List<Vehicle> newVechicles = new List<Vehicle>{ newCar1, newCar2, newCar3, newCar4 };
+            catalog.Vehicles = newVechicles;
+            
+            try
+            {
+                catalog.PrintVehicles(newVechicles);
+                Assert.IsTrue (true);
+            }
+            catch
+            {
+                Assert.IsTrue(false);
+            }
+        }
+
+        [Test]
+        public void Test_ShouldReturnEmptyListWhenCatalogIsEmpty()
+        {
+            //Arange
+            CatalogueVehicle catalog = new CatalogueVehicle();
+            catalog.Vehicles = new List<Vehicle>();
+
+            //Act
+            var result = catalog.GetVehiclesByCriteria("accending");
+            List<Vehicle> newListAccending = new List<Vehicle>();
+
+            //Assert
+            CollectionAssert.AreEqual(newListAccending, result);
+        }
+        [Test]
+        public void Test_ShouldReturnListAccendingOrder()
+        {
+            //[x] Да измисля как да вкарам обектите във листа 
+            //[x] Масив или Лист само със брандове
+            //[x] Сравнявам резълта дали ще съвпадат със листа на ръка 
+
+            //Arange
+
+            Car newCar1 = new Car("Car", "Mercedes", "CLK500", "250");
+            Car newCar2 = new Car("Car", "Audi", "A3", "90");
+            Car newCar3 = new Car("Car", "Opel", "Astra", "150");
+            Car newCar4 = new Car("Car", "BMW", "M5", "350");
+
+            CatalogueVehicle catalog = new CatalogueVehicle();
+
+            catalog.Vehicles = new List<Vehicle> { newCar1, newCar2, newCar3, newCar4 };
+            List<Vehicle> newListAccending = new List<Vehicle> { newCar2, newCar4, newCar1, newCar3 };
+
+            //Act
+            var resultAccending = catalog.GetVehiclesByCriteria("accending");
+
+            //Assert
+            CollectionAssert.AreEqual(newListAccending, resultAccending);
+        }
+
+        [Test]
+        public void Test_ShouldReturnListDescendingOrder()
+        {
+            //Arange
+
+            Car newCar1 = new Car("Car", "Mercedes", "CLK500", "250");
+            Car newCar2 = new Car("Car", "Audi", "A3", "90");
+            Car newCar3 = new Car("Car", "Opel", "Astra", "150");
+            Car newCar4 = new Car("Car", "BMW", "M5", "350");
```

Comment:
Can this logic be extracted, as it is common for most tests? I guess it could be 'reinitialized' before each test

Reply:
Can this be done in the Setup?